### DISCUSSION TITLE
Installation on premise out of docker.

### DISCUSSION THREAD

**Discussion by buzzzo** [hoophq/hoop#540] [General]
Hi 

I didn't find any info about running and installing hoop.dev on premise without docker.
Is it supposed that it should be run only via docker ?

Thx

**Comment by sandromello** [comment]
At the moment we lack an installer for on premisses, therefore you could use either docker or create your own installer. It should be easy to bootstrap a gateway in a Linux with Systemd, we have done that for our [AWS cloud formation recipe](https://github.com/hoophq/hoop/blob/main/deploy/aws/hoopdev-platform.template.yaml).

See the script below for an example:

```sh
#!/bin/bash

set -eo pipefail

VERSION=1.27.12
# you could use arm64 instead
curl -sL https://releases.hoop.dev/release/$VERSION/hoopgateway_$VERSION-Linux_amd64.tar.gz \
  -o hoopgateway_$VERSION-Linux_amd64.tar.gz

mkdir -p /etc/hoopgateway.d /opt/hoop

# make sure the migration path doesn't have any files
rm -rf /opt/hoop/migrations

tar --extract --file hoopgateway_$VERSION-Linux_amd64.tar.gz -C / --strip 1 && \
  rm -f hoopgateway_$VERSION-Linux_amd64.tar.gz && \
  chown -R root: /opt/hoop

# systemd env configuration file
cat - > /etc/hoopgateway.d/config <<EOF
PLUGIN_AUDIT_PATH=/opt/hoop/sessions
PLUGIN_INDEX_PATH=/opt/hoop/sessions/indexes
STATIC_UI_PATH=/opt/hoop/webapp/public
MIGRATION_PATH_FILES=/opt/hoop/migrations
LOG_LEVEL=info
LOG_ENCODING=json
API_URL=https://your-public-address
GRPC_URL=https://your-public-address:8443
POSTGRES_DB_URI=postgres://pguser:pgpasswd@database-endpoint:5432/postgres
IDP_ISSUER=idp-address-issuer
IDP_CLIENT_ID=oauth2-client-id
IDP_CLIENT_SECRET=oauth2-client-secret
EOF

# systemd service
cat > /lib/systemd/system/hoopgateway.service <<EOF
[Unit]
Description=Hoop Gateway
After=network.target

[Service]
ExecStart=/opt/hoop/bin/hoop start gateway
Environment=PATH=/opt/hoop/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
EnvironmentFile=/etc/hoopgateway.d/config
Restart=on-failure
RestartSec=15s

[Install]
WantedBy=multi-user.target
EOF

# enable and start the gateway
systemctl enable hoopgateway && systemctl start hoopgateway
```

- It's important to configure the database properly before starting the gateway, see our [database configuration guide in the Kubernetes section](https://hoop.dev/docs/getting-started/installation/kubernetes#database-configuration)
- We recommend using a persistent volume for `/opt/hoop/sessions` path
- For upgrades it should be safe to just download a new version and restart the gateway, but make sure to follow the releases to see if there's any structural changes.

**Comment by buzzzo** [comment]
Thx for your response.
In the end i would go to docker.

Thx
